### ISSUE TITLE
Add script to disable OneDrive nonsense?

### ISSUE THREAD

**Issue by wjk** [Raphire/Win11Debloat#9]
First of all, thank God for people like you who make these scripts. I now have something to add to my Windows setup automation toolbox (which is annoying enough to use already, but that’s not your fault).

I am not certain if you have heard, but Microsoft has enshittified OneDrive via making it impossible to turn off Desktop and Documents folder sync, and have it stay off. The article I read (paywalled, unfortunately) says that there will be a Group Policy setting to handle this. I do not know what it is or if it has been released yet.

I just wanted to ensure that this is on your radar.

**Comment by Raphire** [comment]
Heya,

Thanks for bringing the Onedrive issue to my attention. It really is a shame how MS is ruining their OS by forcing features on people :/

Are you looking for a solution to disable/uninstall onedrive completely, or just the folder syncing part?

**Comment by wjk** [comment]
Both, honestly? 🤔 I would appreciate both a feature to disable OneDrive completely (to be put in the custom list) and to disable the folder syncing part (to be put in the default list).

**Comment by Raphire** [comment]
Fair enough! 

I'll have to look into this more before I can implement this though, since it affects people's files, especially disabling syncing which moves the folder locations around.

**Comment by reemoteg** [comment]
Sorry if this is a dumb question, but I'm new to Win11 and it's melting my brain... disabling the OneDrive nonsense would also disable the Windows Backup "feature," correct?  BTW, thank you so much for your work.

**Comment by Raphire** [comment]
> Sorry if this is a dumb question, but I'm new to Win11 and it's melting my brain... disabling the OneDrive nonsense would also disable the Windows Backup "feature," correct? BTW, thank you so much for your work.

This is true, although some people might want to use onedrive without using the folder backup feature, so giving the option to just disable the folder backup might be useful to some people

**Comment by reemoteg** [comment]
Agreed on having the option.  Personally, I would disable both for office workstations, but I can see why someone might still want to have the option to use onedrive without the backup feature.  Thank you again.

**Comment by ZaxLofful** [comment]
I agree with it being an option, but it should be strictly that "optional" and not part of the default config

**Comment by Raphire** [comment]
> I agree with it being an option, but it should be strictly that "optional" and not part of the default config

This would definitely be an option, it will not be added to the default config.

**Comment by Raphire** [comment]
Just a short update on this. I've been trying to find a good, consistent way to remove OneDrive, but unfortunately it seems to be quite a mess. Sure, there's plenty of articles with all kinds of solutions for this issue, but I haven't been able to find a method that works well and is suitable for this script.

Disabling OneDrive folder backups does look more promising, I'll be testing this soon and I'll update this issue if I have more info.

P.S. If you have a good way to uninstall OneDrive via powershell or any other way (apart from manually uninstalling it) please let me know :)

**Comment by Raphire** [comment]
The ability to uninstall OneDrive has been added in 0d7910e

**Comment by garys-primary** [comment]
Just wanted to share - uninstalled OneDrive myself first Windows launch - right after Windows warned that file versions has changed and been restored in my `Documents\git\` folder.  

After uninstalling git, I disabled the OneDrive with group policy. Those Documents, Images, Videos etc shortcuts in Explorer instantly broke. I had to unpin them then delete and recreate new Documents, Images etc folders. Then changed paths in registry - which had a lot of OneDrive paths.

Some x11-started programs like Blender and Inkskape had issues with that later, but after full reinstall it got fixed
